frontend/packages/llama-stack-modular-ui/bff/internal/integrations/kubernetes/token_k8s_client.go

```diff
@@ -103,6 +135,28 @@ func (kc *TokenKubernetesClient) GetNamespaces(ctx context.Context, _ *integrati
 	return nsList.Items, nil
 }
 
+func (kc *TokenKubernetesClient) GetLlamaStackDistributions(ctx context.Context, identity *integrations.RequestIdentity, namespace string) (*lsdapi.LlamaStackDistributionList, error) {
+	ctx, cancel := context.WithTimeout(ctx, 30*time.Second)
+	defer cancel()
+
+	lsdList := &lsdapi.LlamaStackDistributionList{}
+
+	listOptions := &client.ListOptions{
+		Namespace: namespace,
+	}
+	err := kc.Client.List(ctx, lsdList, listOptions)
+	if err != nil {
+		kc.Logger.Error("failed to list LlamaStackDistributions", "error", err, "namespace", namespace)
+
+		if errors.IsNotFound(err) {
```

Comment:
why not return the error as-is ? or at least be sure that we return an error that is also a `IsNotFound` ?

Reply:
I have modified this to return the error as-is, and let the handler differentiate it into whether it's a bad request, or a malformed request.